task2.js

```diff
@@ -0,0 +1,17 @@
+const csv = require("csvtojson");
+const fsPromise = require("fs/promises");
+const path = require("path");
+
+csv()
+  .fromFile("csv/nodejs-hw1-ex1.csv")
+  .then((jsonObj) => {
+    console.log(JSON.stringify(jsonObj));
+    fsPromise
+      .writeFile(
+        path.resolve(__dirname, "result.txt"),
+
+        JSON.stringify(jsonObj, null, 4)
+      )
+      .then(() => console.log("File was converted"))
+      .catch((error) => console.log(error));
+  });
```

Comment:
Please do it via stream.pipeline (read about it here: https://nodejs.org/api/stream.html#stream_stream_pipeline_streams_callback)
It is better to use for read, transform, write streams.

Reply:
That's why we have this educational program☺

Please fix everywhere, in all files, so it looks good everywhere. By the way, you did it correctly, but I wanted you to do it even more elegant way via (stream pipeline) as mentioned previously.

Please view this exact documentation about [streampipeline](https://nodejs.org/api/stream.html#streampipelinestreams-callback)
and look at code example with **asynchronous run** function. And please try to do it the same way by understanding how easily you can play with streams.

Reply:
![image](https://user-images.githubusercontent.com/73526360/207565182-717f5b9c-de6e-46d8-8f16-1e4a1930e570.png)
now everything look something like this)
Is understood correctly first of all the `pipeline` newer feature, `pipe` older, therefore new feature more powerful))) 👍 
pipe is method of Readable whereas `pipeline` a module method to pipe between streams forwarding errors and properly cleaning up and provide a callback when the pipeline is complete. Looks like `pipeline` prevent memory leak. 
I am not sure, but maybe the most significant benefit callback for catch error.

Reply:
![image](https://user-images.githubusercontent.com/73526360/207566660-5ee0992d-2bb0-45da-a5c1-849b93eea4cb.png)
liked this phrase from this article. (https://dev.to/yenyih/use-pipeline-instead-of-pipe-4gl3)

Reply:
That looks better. Last thing, cosmetics, can you please extract readable and writable stream as constants. And please extract function for handling final results or processing error.

And that would be enough.

Reply:
Thanks now look better
![image](https://user-images.githubusercontent.com/73526360/207572461-eb2039fb-7783-4a95-9a94-40221490619e.png)